ddht/v5_1/network.py

```diff
@@ -171,18 +173,14 @@ async def talk(
         )
         return response.message.payload
 
-    async def get_enr(
+    async def lookup_enr(
         self, node_id: NodeID, *, enr_seq: int = 0, endpoint: Optional[Endpoint] = None
     ) -> ENRAPI:
-        try:
-            enr = self.enr_db.get_enr(node_id)
-        except KeyError:
+        enr = self.enr_db.get_enr(node_id)
```

Comment:
So I think we want to revert to the previous logic and instead do it as:

```python
            enr = self.enr_db.get_enr(node_id)

        except KeyError:
            if endpoint is None:
                raise  # if we don't already know the `endpoint` it isn't possible for us to send a request.
            enr = await self._fetch_enr(node_id, endpoint=endpoint)
            self.enr_db.set_enr(enr)
```

The idea here is that we *can* issue the request if we happen to already know the `node_id` & `endpoint` and in that case, we have a path to success even if we have no entry in our database.

Reply:
ahh, great point. fixed